src/storage/src/hummock/compactor/compaction_executor.rs

```diff
@@ -13,72 +13,65 @@
 // limitations under the License.
 
 use std::future::Future;
-use std::pin::Pin;
 
-use crate::hummock::compactor::CompactOutput;
-use crate::hummock::{HummockError, HummockResult};
-
-type CompactionRequest = Pin<Box<dyn Future<Output = ()> + Send + 'static>>;
+use tokio::task::JoinHandle;
 
 /// `CompactionExecutor` is a dedicated runtime for compaction's CPU intensive jobs.
+#[cfg(not(madsim))]
 pub struct CompactionExecutor {
-    requests: tokio::sync::mpsc::UnboundedSender<CompactionRequest>,
-    // TODO: graceful shutdown
-    #[cfg(not(madsim))]
-    _runtime_thread: std::thread::JoinHandle<()>,
+    runtime: Option<tokio::runtime::Runtime>,
 }
 
+#[cfg(not(madsim))]
 impl CompactionExecutor {
-    #[cfg(not(madsim))]
     pub fn new(worker_threads_num: Option<usize>) -> Self {
-        let (tx, mut rx) = tokio::sync::mpsc::unbounded_channel();
+        let mut builder = tokio::runtime::Builder::new_multi_thread();
+        if let Some(worker_threads_num) = worker_threads_num {
```

Comment:
if we had not set the worker_threads_num,  the `tokio` runtime will use `the number of cores` as default , is it expected ?

It's possible that we run both `compactor` and `compute node` on the same `instance`, and the `compactor` dedicated runtime will run out of cpu resources

Reply:
No, if compactor and compute node run on the same instance, it will only use small thread.